### PR TITLE
Add Crossing Pedestrian Detection example

### PR DESCRIPTION
**Pull request by sandalns**
### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: examples/dataset/crossing_pedestrian_detection/crossing_pedestrian_detection/upload_results.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+                "frame_metrics": frame_metrics_lst,
+                "TP": final_tps,
+                "FP": final_fps,
+                "FN": final_fns,
+                "TN": final_tns,
+                "has_TP": True if len(final_tps) > 0 else False,
+                "has_FP": True if len(final_fps) > 0 else False,
+                "has_FN": True if len(final_fns) > 0 else False,
+                "has_TN": True if len(final_tns) > 0 else False,
+                "CrossFramePrecision": np.mean(cross_frame_precisions),
```
Comment (bagalaster) [inline examples/dataset/crossing_pedestrian_detection/crossing_pedestrian_detection/upload_results.py:272]: Just curious - where did these "cross frame" metrics come from? Is this a canonical method in object tracking evaluation, or did we find some issue with the usual object detection metrics that would be derived from TP/FP/FN above?

Comment (sandalns) [reply examples/dataset/crossing_pedestrian_detection/crossing_pedestrian_detection/upload_results.py:272]: These metrics came up in past discussions with @marko-polo-cheno as the most valuable metrics to highlight. The idea that the mean of precision, recall and f1 across all frames within a video is more meaningful than taking the total TP/FP/FN and then calculating precision, recall and f1.

File: examples/dataset/crossing_pedestrian_detection/crossing_pedestrian_detection/upload_results.py [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+                    time_to_event = predictions[bbox.ped_id][bbox.frame_id]["tte"]
+
+                scored_bboxes_lst.append(
+                    ScoredPedestrianBoundingBox(
+                        ped_id=bbox.ped_id,
+                        top_left=bbox.top_left,
+                        bottom_right=bbox.bottom_right,
+                        frame_id=bbox.frame_id,
+                        occlusion=bbox.occlusion,
+                        time_to_event=time_to_event if not failed_to_infer else None,
```
Comment (yifanwu-kolena) [inline examples/dataset/crossing_pedestrian_detection/crossing_pedestrian_detection/upload_results.py:134]: based on [this example](https://trunk.kolena.io/nitesh-dev/dataset?datasetId=67), a lot of the FN, TN bboxes have no time_to_event or null value, is this expected?

**Review by yifanwu-kolena — APPROVED** [review]

File: examples/dataset/crossing_pedestrian_detection/crossing_pedestrian_detection/upload_results.py (hunk 2 above)
Comment (sandalns) [reply examples/dataset/crossing_pedestrian_detection/crossing_pedestrian_detection/upload_results.py:134]: Yes, the FNs and TNs are ground truth bboxes which do not have these fields.

**Pull request merged**